internal/resource/equality.go

```diff
@@ -0,0 +1,37 @@
+package resource
```

Comment:
These should really be generated by a `protoc` plugin, but I couldn't find one that was well-maintained.